packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -217,124 +179,77 @@ const DataGrid = injectIntl((props) => {
     props.focusFuncRef,
     () => ({
       setFocusedRowCol,
-      getFocusedCell() { return { row: focusedRow, col: focusedCol }; },
+      getFocusedCell() { return { row: focusedRow.current, col: focusedCol.current }; },
     }),
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    [focusedCol, focusedRow],
+    [focusedCol.current, focusedRow.current],
   );
 
   // -------------------------------------
   // callback Hooks
 
-  const gridRef = useCallback((node) => {
-    if (!node) {
-      return;
-    }
-
-    grid.current = node;
-
-    const resizeObserver = new ResizeObserver(() => {
-      // Update table height state variable
-      setTableHeight(grid.current.offsetHeight - 1);
-    });
-
-    // Register resize observer to detect size changes
-    resizeObserver.observe(node);
-  }, []);
-
   // -------------------------------------
   // useEffect Hooks
 
-  // useEffect for row selection
   useEffect(() => {
-    setDataGridColumns(displayedColumns.map((column) => initializeColumn(column)));
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [hasSelectableRows]);
-
-  // useEffect for row displayed columns
-  useEffect(() => {
-    setDataGridColumns(displayedColumns.map((column) => initializeColumn(column)));
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [pinnedColumns, overflowColumns]);
-
-  // useEffect to calculate pinned column offsets
-  useEffect(() => {
-    const offsetArray = [];
-    let cumulativeOffset = 0;
-    let lastPinnedColumnIndex;
-
-    // if grid has selectable rows but no pinned columns, then set the offset of the first column to 0
-    if (hasSelectableRows && pinnedColumns.length === 0) {
-      lastPinnedColumnIndex = 0;
-      offsetArray.push(cumulativeOffset);
-      setPinnedColumnOffsets(offsetArray);
-      return;
+    if (gridContainerRef) {
+      const table = gridContainerRef.current.getElementsByTagName('table')[0];
+      tableRef.current = table;
     }
```

Comment:
Instead of querying the DOM, shouldn't the Table be able to accept a prop where this ref can be set?  I would not think that we would need to query the DOM to get this information.

Reply:
Had a discussion, it turns out that we can't use forwardRef with components wrapped with react-intl: https://formatjs.io/docs/react-intl/upgrade-guide-3x/#migrate-withref-to-forwardref

So we'll just have to query the DOM. It's not the ideal solution but we can hopefully solve this unideal behavior with our tech currency initiative.